File: apps/backend/app/Application/UseCases/ServiceLog/CreateServiceLogUseCase.php
<?php

namespace App\Application\UseCases\ServiceLog;

use App\Application\DTOs\ServiceLog\CreateServiceLogDTO;
use App\Domain\ServiceLog\Contracts\ServiceLogRepositoryInterface;
use App\Domain\ServiceLog\Entities\ServiceLog;
use Illuminate\Support\Str;

class CreateServiceLogUseCase
{
    public function __construct(
        private ServiceLogRepositoryInterface $serviceLogRepository,
    ) {}

    public function execute(CreateServiceLogDTO $dto): ServiceLog
    {
        $now = new \DateTimeImmutable();

        $serviceLog = new ServiceLog(
            id: (string) Str::uuid(),
            tenantId: $dto->tenantId,
            clientResourceId: $dto->clientResourceId,
            serviceId: $dto->serviceId,
            reservationId: $dto->reservationId,
            attendedBy: $dto->attendedBy,
            createdBy: $dto->createdBy,
            startedAt: $now,
            finishedAt: null,
            priceCharged: $dto->priceCharged,
            paymentMethod: $dto->paymentMethod,
            status: 'in_progress',
            notes: $dto->notes,
            logDate: $now->format('Y-m-d'),
        );

        return $this->serviceLogRepository->save($serviceLog);
    }
}

File: apps/admin_mobile/lib/application/use_cases/service_logs/create_service_log_use_case.dart
import '../../../domain/entities/service_log.dart';
import '../../../domain/repositories/service_log_repository.dart';

class CreateServiceLogUseCase {
  final ServiceLogRepository _repo;
  CreateServiceLogUseCase(this._repo);

  Future<ServiceLog> call({
    required int clientResourceId,
    required int serviceId,
    required int attendedBy,
    required double priceCharged,
    required PaymentMethod paymentMethod,
    String? notes,
  }) =>
      _repo.create(
        clientResourceId: clientResourceId,
        serviceId: serviceId,
        attendedBy: attendedBy,
        priceCharged: priceCharged,
        paymentMethod: paymentMethod,
        notes: notes,
      );
}
